app/src/main/java/ch/hikemate/app/ui/components/AsyncStateHandler.kt

```diff
@@ -0,0 +1,47 @@
+package ch.hikemate.app.ui.components
+
+import android.util.Log
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Home
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.ui.res.stringResource
+
+/**
+ * A component that handles the state of an asynchronous operation and displays the appropriate
+ * content based on the state. If an error occurred, an error message is displayed. If the value is
+ * null, a loading animation is displayed. Otherwise, the content is displayed.
+ *
+ * @param errorMessageIdState The state that holds the error message id
+ * @param actionContentDescriptionStringId The string id for the content description of the action
+ * @param actionOnErrorAction The action to perform when the error action is clicked
+ * @param valueState The state that holds the value to display
+ * @param content The content to display
+ */
+@Composable
```

Comment:
Comment: This function seems pretty good, I've tested again on my phone and by reviewing the code it seems equivalent to the code you had before, but much better, since there isn't any code repetition.
You also did well by treating the cases whenver the user is not signed in and whenever there is an error this makes debuging much easier.